public/components/JSONDiffEditor/JSONTextArea.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { forwardRef, useState, useEffect, useImperativeHandle, useRef } from "react";
+import { EuiFormRow } from "@elastic/eui";
+import type { MonacoDiffEditorProps } from "react-monaco-editor";
+import { IJSONEditorRef } from "../JSONEditor";
+
+export interface JSONDiffEditorProps extends Partial<MonacoDiffEditorProps> {
+  value: string;
+  onChange?: (value: JSONDiffEditorProps["value"]) => void;
+  "data-test-subj"?: string;
+  disabled?: boolean;
+}
+
+const JSONDiffEditor = forwardRef(({ value, onChange, ...others }: JSONDiffEditorProps, ref: React.Ref<IJSONEditorRef>) => {
+  const [confirmModalVisible, setConfirmModalVisible] = useState(false);
+  const [editorValue, setEditorValue] = useState(value);
+  const valueRef = useRef(editorValue);
+  valueRef.current = editorValue;
```

Comment:
why do we need this line if we already set the initial value in above line?

Reply:
There is a closure problem for react hooks, you can see there is a `getValue` method which gives the latest value of the editor, if we do not set it to a ref, when we call ref.getValue the result will always be the initial value.